crates/turborepo-ui/src/tui/preferences.rs

```diff
@@ -1,132 +1,73 @@
-
-        // TODO: Is this really how to do this? No way, right?
-        match field {
-            PreferenceFields::IsTaskListVisible => {
-                json["is_task_list_visible"] = new_value;
-            }
-            PreferenceFields::ActiveTask => {
-                json["active_task"] = new_value;
-            }
-            PreferenceFields::PinnedTaskSelection => {
-                json["is_pinned_task_selection"] = new_value;
-            }
-        }
 
-        let updated_json_string = serde_json::to_string_pretty(&json)?;
-
-        let mut file = fs::File::create(&preferences_file)?;
-        file.write_all(updated_json_string.as_bytes())?;
-
-        Ok(())
+    pub fn is_task_list_visible(&self) -> bool {
+        self.config.is_task_list_visible.unwrap_or(true)
     }
 
-    pub fn read_pinned_task_state(repo_root: &AbsoluteSystemPathBuf) -> bool {
-        let preferences_file = repo_root.join_components(TUI_PREFERENCES_PATH_COMPONENTS);
-
-        read_json(&preferences_file)
-            .is_pinned_task_selection
-            .unwrap_or(false)
+    pub fn active_task(&self) -> Option<&str> {
+        let active_task = self.config.active_task.as_deref()?;
+        Some(active_task)
     }
 
-    pub fn read_task_list_visibility(repo_root: &AbsoluteSystemPathBuf) -> bool {
-        let preferences_file = repo_root.join_components(TUI_PREFERENCES_PATH_COMPONENTS);
-
-        read_json(&preferences_file)
-            .is_task_list_visible
-            .unwrap_or(true)
+    pub fn set_is_task_list_visible(&mut self, value: Option<bool>) -> Result<(), Error> {
+        self.config.is_task_list_visible = value;
+        self.flush_to_disk()
     }
 
-    pub fn get_selected_task_index(
-        repo_root: &AbsoluteSystemPathBuf,
-        tasks_by_status: &TasksByStatus,
-    ) -> usize {
-        let preferences_file = repo_root.join_components(TUI_PREFERENCES_PATH_COMPONENTS);
+    pub fn set_active_task(&mut self, value: Option<String>) -> Result<(), Error> {
+        self.config.active_task = value;
+        self.flush_to_disk()
+    }
 
-        let selected_task_name = read_json(&preferences_file)
-            .active_task
-            .unwrap_or("".to_string());
+    fn flush_to_disk(&self) -> Result<(), Error> {
+        self.file_path.ensure_dir()?;
+        self.file_path
+            .create_with_contents(serde_json::to_string_pretty(&self.config)?)?;
+        Ok(())
+    }
+}
 
-        match tasks_by_status
-            .task_names_in_displayed_order()
-            .position(|task_name| *task_name == selected_task_name)
-        {
-            Some(index) => index,
-            None => {
-                let _ = Self::update_preference(
-                    repo_root,
-                    PreferenceFields::PinnedTaskSelection,
-                    serde_json::Value::Bool(false),
-                );
+#[derive(Serialize, Deserialize, Debug)]
+struct Preferences {
+    pub is_task_list_visible: Option<bool>,
```

Comment:
I don't think we need this to be optional unless we want to allow users to mess with the config file manually.